### ISSUE TITLE
false negative type error for assoc in function

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#756]
```
$ clj-kondo --lint - <<< '(defn foo [x] (assoc x :foo true)) (inc (foo {}))'
linting took 36ms, errors: 0, warnings: 0
```
